src/Core/View.js

```diff
@@ -217,11 +234,14 @@ function _preprocessLayer(view, layer, provider) {
  * });
  *
  * // Add Elevation Layer
- * view.addLayer({
+ * var layer = view.addLayer({
  *      type: 'elevation',
  *      id: 'iElevation',
  * });
  *
+ * // React to preprocessing
+ * layer.whenReady.then(() => { ... });
```

Comment:
Maybe you should modify `addLayer` to return `layer.whenReady` instead of `layer`. WDYT?